lib/lambda_ethereum_consensus/state_transition/operations.ex

```diff
@@ -0,0 +1,189 @@
+defmodule LambdaEthereumConsensus.StateTransition.Operations do
+  @moduledoc """
+  State transition Operations related functions
+  """
+
+  alias LambdaEthereumConsensus.StateTransition.Accessors
+  alias LambdaEthereumConsensus.StateTransition.Mutators
+  alias LambdaEthereumConsensus.StateTransition.Predicates
+  alias SszTypes.BeaconState
+  alias SszTypes.ExecutionPayload
+  alias SszTypes.Validator
+  alias SszTypes.Withdrawal
+
+  @doc """
+  Apply withdrawals to the state.
+  """
+  @spec process_withdrawals(BeaconState.t(), ExecutionPayload.t()) ::
+          {:ok, BeaconState.t()} | {:error, String.t()}
+  def process_withdrawals(
+        %BeaconState{
+          validators: validators,
+          next_withdrawal_validator_index: next_withdrawal_validator_index
+        } = state,
+        %ExecutionPayload{withdrawals: withdrawals}
+      ) do
+    expected_withdrawals = get_expected_withdrawals(state)
+    result = decrease_balances(state, withdrawals, expected_withdrawals)
+
+    response =
+      case result do
+        {:ok, state} ->
+          # Update the next withdrawal index if this block contained withdrawals
+          new_state =
+            if length(expected_withdrawals) != 0 do
+              latest_withdrawal = List.last(expected_withdrawals)
+              %Withdrawal{index: index} = latest_withdrawal
+              %BeaconState{state | next_withdrawal_index: index + 1}
+            else
+              state
+            end
```

Comment:
Instead of this let's just 

```elixir
state
|> update_withdrawal_index(length(expected_withdrawals))
|> update_next_validator_index(expected_withdrawals)
```

And then we can have some additional functions:

```elixir
defp update_withdrawal_index(%BeaconState{next_withdrawal_index: idx} = state, processed_withdrawals) do
  # As suggested in eth2book
  %BeaconState{state | next_withdrawal_index: idx + processed_withdrawals}
end

defp update_next_validator_index(state, expected_withdrawals) ...
```

And we can have two clauses for that second function instead of an if.